astroid/interpreter/_import/spec.py

```diff
@@ -349,7 +357,7 @@ def _find_spec_with_path(
     module_parts: list[str],
     processed: list[str],
     submodule_path: Sequence[str] | None,
-) -> tuple[Finder, ModuleSpec]:
+) -> tuple[Finder | MetaPathFinder, ModuleSpec]:
```

Comment:
I'm not so sure we can limit the type of the return object now. The `six._SixMetaPathImporter` inherits directly from `object`:

```py
(Pdb) meta_finder.__class__.mro()
[<class 'urllib3.packages.six._SixMetaPathImporter'>, <class 'object'>]
```